internal/tool/command.go

```diff
@@ -0,0 +1,160 @@
+	semgrepCmd := createCommand(configFile, toolExecution.SourceDir, language, files)
+
+	semgrepOutput, semgrepError, err := runCommand(semgrepCmd)
+	if err != nil {
+		return nil, errors.New("Error running semgrep: " + semgrepError + "\n" + err.Error())
+	}
+
+	output, err := parseCommandOutput(toolExecution.ToolDefinition, patternDescriptions, semgrepOutput)
+	if err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func createCommand(configFile *os.File, sourceDir, language string, files []string) *exec.Cmd {
+	params := createCommandParameters(language, configFile, files)
+	cmd := exec.Command("semgrep", params...)
+	cmd.Dir = sourceDir
+
+	return cmd
+}
+
+func createCommandParameters(language string, configFile *os.File, filesToAnalyse []string) []string {
+	// adding -json parameters
+	cmdParams := []string{
+		"-json", "-json_nodots",
+	}
+	// adding -lang parameter
+	cmdParams = append(
+		cmdParams,
+		"-lang", language,
+	)
+	// adding -rules parameter
+	cmdParams = append(
+		cmdParams,
+		"-rules", configFile.Name(),
+	)
+	// adding files to analyse
+	cmdParams = append(
+		cmdParams,
+		filesToAnalyse...,
+	)
+	return cmdParams
+}
+
+func runCommand(cmd *exec.Cmd) (string, string, error) {
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	cmdOutput, err := cmd.Output()
+	if err != nil {
+		return "", stderr.String(), err
+	}
+	return string(cmdOutput), "", nil
+}
+
+func parseCommandOutput(toolDefinition codacy.ToolDefinition, patternDescriptions *[]codacy.PatternDescription, commandOutput string) ([]codacy.Result, error) {
+	var result []codacy.Result
+	scanner := bufio.NewScanner(strings.NewReader(commandOutput))
+	for scanner.Scan() {
+		output := scanner.Text()
+		result = appendToResult(result, patternDescriptions, output)
+	}
+
+	return result, nil
+}
+
+func appendToResult(result []codacy.Result, patternDescriptions *[]codacy.PatternDescription, output string) []codacy.Result {
+
+	var semgrepOutput SemgrepOutput
+	json.Unmarshal([]byte(output), &semgrepOutput)
+	result = appendIssueToResult(result, patternDescriptions, semgrepOutput)
+	result = appendErrorToResult(result, semgrepOutput)
+	return result
+}
+
+func appendIssueToResult(result []codacy.Result, patternDescriptions *[]codacy.PatternDescription, semgrepOutput SemgrepOutput) []codacy.Result {
+	for _, semgrepRes := range semgrepOutput.Results {
+		result = append(result, codacy.Issue{
+			PatternID:  semgrepRes.CheckID,
+			Message:    writeMessage(patternDescriptions, semgrepRes.CheckID, strings.TrimSpace(semgrepRes.Extra.Message)),
+			Line:       semgrepRes.StartLocation.Line,
+			File:       semgrepRes.Path,
+			Suggestion: semgrepRes.Extra.RenderedFix,
+		})
+	}
+
+	return result
+}
+
+func writeMessage(patternDescriptions *[]codacy.PatternDescription, ID string, s string) string {
```

Comment:
```suggestion
func getMessage(patternDescriptions *[]codacy.PatternDescription, id, s string) string {
```

Rename `s` to something that makes more sense.